taraxa/state/dpos/precompiled/dpos_contract.go

```diff
@@ -323,6 +332,16 @@ func (self *Contract) delegate(ctx vm.CallFrame, block types.BlockNum, args Vali
 	if validator == nil {
 		return ErrNonExistentValidator
 	}
+
+	if self.cfg.MaximumStake.Cmp(bigutil.Big0) != 0 && self.cfg.MaximumStake.Cmp(bigutil.Add(ctx.Value, validator.TotalStake)) == -1 {
+		return ErrValidatorsMaxStakeExceeded
+	}
+
+	delegation := self.delegations.GetDelegation(ctx.CallerAccount.Address(), &args.Validator)
+	if delegation == nil && self.cfg.MinimumDeposit.Cmp(bigutil.Big0) != 0 && self.cfg.MinimumDeposit.Cmp(ctx.Value) == 1 {
```

Comment:
not good. this must be checked even for delegation != nil because you might end up with delegation < MIN due to slashing so next time when you delegate the total amount must be >= MIN

Reply:
nope, slashing only happens when you undelegate

Reply:
that's probably issue we have

Reply:
what sense does it make to check MinimumDeposit to 0 ? Even if MinimumDeposit == 0, then this condition is enough: 
self.cfg.MinimumDeposit.Cmp(ctx.Value) == 1

Reply:
you are right, good point

Reply:
So I can be malicious and do bad things all the time, never undelegate and we never decrease my delegation ? if slashing happens only when you undelegate then it is not slashing - probably cant work like that.